http/tokenV2/middleware.go

```diff
@@ -138,18 +177,95 @@ func credentialIsSecure(credential string) error {
 	return fmt.Errorf("no signatures found")
 }
 
+// bestPracticesCheck ensures tokens are crafted in a sensible way, ensuring that even a valid signer must
+// conform to certain security controls such as not creating long lived credentials etc.
+//
+// WARNING: This check does not verify signatures or even the essential claims such as NotBefore, Expiration, etc.
+func bestPracticesCheck(token jwt.Token) error {
+	// Ensure the IssuedAt (iat) field is present
+	if _, ok := token.Get(jwt.IssuedAtKey); !ok {
```

Comment:
wondering; why is `iat` relevant? (if we derive validity from `nbf` and `exp`)

Reply:
This is to prevent somebody issuing a sort of "time bomb" token, which is only valid for a few hours but takes a long time to become valid. It's something that in fact is quite strange to do but could look fairly innocuous if you only inspect exp-nbf.